src/optimizer/rule/ordered_aggregate_optimizer.cpp

```diff
@@ -24,6 +30,90 @@ unique_ptr<Expression> OrderedAggregateOptimizer::Apply(LogicalOperator &op, vec
 		changes_made = true;
 		return nullptr;
 	}
+
+	auto &context = rewriter.context;
+	if (context.config.enable_optimizer) {
+		// for each ORDER BY - check if it is actually necessary
+		// expressions that are in the groups do not need to be ORDERED BY
+		// `ORDER BY` on a group has no effect, because for each aggregate, the group is unique
+		// similarly, we only need to ORDER BY each aggregate once
+		expression_set_t seen_expressions;
+		auto &groups = op.Cast<LogicalAggregate>().groups;
+		for (auto &target : groups) {
+			seen_expressions.insert(*target);
+		}
+		vector<BoundOrderByNode> new_order_nodes;
```

Comment:
Could we add some tests that trigger/stress-test this optimization?

Reply:
We already do - I added it because of some failures!